tests/test_sortedqueue.py

```diff
@@ -0,0 +1,165 @@
+        self.pq.push(0)
+        self.pq.push(1)
+        self.pq.push(2)
+        self.pq.push(3)
+        self.pq.push(4)
+        self.pq.push(5)
+        self.pq.push(6)
+
+        assert self.pq.popiv() == (3, 26)
+        assert self.pq.popiv() == (0, 31)
+        assert self.pq.popiv() == (1, 41)
+        assert self.pq.popiv() == (4, 53)
+        assert self.pq.popiv() == (5, 58)
+        assert self.pq.popiv() == (2, 59)
+        assert self.pq.popiv() == (6, 97)
+
+    def test_push_pop_random(self):
+        """Test pushing and popping random values from the sorted queue."""
+        values = np.random.randint(0, 100, 1024, dtype=np.int32)
+        self.pq = SortedQueue(len(values), values)
+        for i in range(len(values)):
+            self.pq.push(i)
+        minimum = 0
+        while len(self.pq) > 0:
+            value = self.pq.popv()
+            assert value >= minimum
+            minimum = value
+
+    def test_peek(self):
+        """Test peeking at the top element of the sorted queue."""
+        self.pq.push(0)
+        self.pq.push(1)
+        self.pq.push(2)
+        self.pq.push(3)
+        self.pq.push(4)
+        self.pq.push(5)
+        self.pq.push(6)
+
+        assert self.pq.peekiv() == (3, 26)
+
+    def test_empty_peek(self):
+        """Test peeking at the top element of an empty sorted queue. Should raise an IndexError."""
+        with pytest.raises(IndexError):
+            _ = self.pq.peekiv()
+
+    def test_peek_random(self):
+        """Test peeking at the top element of the sorted queue with random values."""
+        values = np.random.randint(0, 100, 1024, dtype=np.int32)
+        self.pq = SortedQueue(len(values), values)
+        for i in range(len(values)):
+            self.pq.push(i)
+        minimum = values.min()
+        assert self.pq.peekv() == minimum
+
+    def test_empty_pop(self):
+        """Test popping from an empty sorted queue. Should raise an IndexError."""
+        with pytest.raises(IndexError):
+            self.pq.pop()
+
+    def test_full_push(self):
+        """Test pushing to a full sorted queue. Should raise an IndexError."""
+        self.pq.push(0)
+        self.pq.push(1)
+        self.pq.push(2)
+        self.pq.push(3)
+        self.pq.push(4)
+        self.pq.push(5)
+        self.pq.push(6)
+
+        with pytest.raises(IndexError):
+            self.pq.push(7)
+
+    def test_push_timing(self):
+        """Test the timing of the push method."""
+        import timeit
+
+        np.random.seed(20240701)
+        count = 1 << 20
+        values = np.random.randint(0, 100, count, dtype=np.int32)
+        self.pq = SortedQueue(len(values), values)
+        elapsed = timeit.timeit("for i in range(len(values)): self.pq.push(i)", globals={"values": values, "self": self}, number=1)
+        messages.append(
+            f"SortedQueue.push() timing: {elapsed:0.4f} seconds for {count:9,} elements = {int(round(count / elapsed)):11,} elements/second"
```

Comment:
I don't think I saw this output in my console when I ran that test.

Reply:
It's a little bit of work to make sure the test harness doesn't swallow up stdout. Should be fixed now.